pyomo/contrib/fbbt/fbbt.py

```diff
@@ -1120,6 +1120,29 @@ def visiting_potential_leaf(self, node):
         if node.is_variable_type():
             lb, ub = self.bnds_dict[node]
 
+            lb, ub = self.bnds_dict[node]
+            if lb > ub:
+                if lb - self.feasibility_tol > ub:
+                    raise InfeasibleConstraintException('Lower bound ({1}) computed for variable {0} is larger than the computed upper bound ({2}).'.format(node, lb, ub))
+                else:
+                    lb -= self.feasibility_tol
+                    ub += self.feasibility_tol
+                    orig_lb = value(node.lb)
+                    orig_ub = value(node.ub)
+                    if orig_lb is None:
+                        orig_lb = -interval.inf
+                    if orig_ub is None:
+                        orig_ub = interval.inf
+                    if lb < orig_lb:
+                        lb = orig_lb
+                    if ub > orig_ub:
+                        ub = orig_ub
+                    self.bnds_dict[node] = (lb, ub)
```

Comment:
This procedure looks good, however a comment here that indicates the logic behind this code would be helpful.  When I initially read this, I was wondering why you were relaxing both `lb` and `ub` ... but the checks against the original bounds make sense (that is, I read this as if the bounds ever cross, then relax the computed values by `feasibility_tol`).